device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java

```diff
@@ -498,26 +498,31 @@ public void close(IotHubConnectionStatusChangeReason reason, Throwable cause) th
             this.taskScheduler.shutdown();
         }
 
-        if (this.iotHubTransportConnection != null)
+        try
         {
-            this.iotHubTransportConnection.close();
+            if (this.iotHubTransportConnection != null)
+            {
+                this.iotHubTransportConnection.close();
+            }
         }
+        finally
+        {
+            this.updateStatus(IotHubConnectionStatus.DISCONNECTED, reason, cause);
```

Comment:
On the off chance that an unchecked exception is thrown from the transport layer, we still need to update the local state accordingly

Reply:
And wake up the worker threads so they can finish

Reply:
what kind of unchecked exceptions might be expected to be thrown here?

Reply:
Possibly "out of memory" type exceptions